evm_arithmetization/src/cpu/kernel/asm/core/precompiles/kzg_peval.asm

```diff
@@ -66,10 +66,8 @@ global store_kzg_verification:
     PUSH @SEGMENT_RETURNDATA
     %build_address_no_offset
     // stack: addr, res_lo, res_hi, kexit_info
-    DUP1 %add_const(32)
```

Comment:
I don't understand how you end up having a stack of the same size if you're only eliminating one op that pushes.

Reply:
(Unless it was not working before)

Reply:
`MSTORE_32BYTES_32` returns the updated offset post write, which wasn't accounted for in the previous stack comment :)